packages/compartment-mapper/src/types/external.ts

```diff
@@ -426,3 +451,8 @@ export type ParseFn = { isSyncParser?: true } & ((
  * ParserImplementations}
  */
 export type ParserForLanguage = Record<Language | string, ParserImplementation>;
+
+/**
+ * Generic logging function accepted by various functions.
+ */
+export type LogFn = (message: string, ...args: any[]) => void;
```

Comment:
I think the variadic args can be more constrained, possibly to types expressible in JSON only, though I acknowledge that the signature of `JSON.parse` returns `any` and, if it can’t be held to a higher standard, why should we hold ourselves that high. In any case, your discretion whether to punt.

Reply:
my take is this: if we end up emitting something that `JSON.stringify` cannot stringify and it breaks somebody, then we can worry about it (since we'd need to implement multiple types to handle it). at that point we can also decide whether to force the data to be immutable or a deep clone or a struct-like object or something (if leaking mutable objects is indeed a concern).

afaik the only thing that `JSON.stringify` would ever tantrum about is an object containing circular references. oftentimes, logging implementations will guard against circular refs anyway.